scripts/braid-fixture.ts

```diff
@@ -0,0 +1,99 @@
+import path from 'path';
+
+import dedent from 'dedent';
+import fse from 'fs-extra';
+import yargs from 'yargs';
+
+import { run as _run, done } from './utils';
+
+const argv = await yargs(process.argv.slice(2))
+  .option('branch', {
+    default: 'master',
+  })
+  .option('clone', {
+    boolean: true,
+    default: false,
+  })
+  .option('absorb', {
+    boolean: true,
+    default: false,
+  })
+  .option('test', {
+    boolean: true,
+    default: false,
+  })
+  .help()
+  .showHelp()
+  .parse();
+console.log();
+
+const __dirname = path.dirname(new URL(import.meta.url).pathname);
+
+const repo = 'git@github.com:seek-oss/braid-design-system.git';
+const submodule = 'fixtures/braid-design-system';
+const branch = argv.branch;
+
+const fromRoot = (location: string) => path.join(__dirname, '..', location);
+const run: typeof _run = (command) => _run(command, { cwd: fromRoot('.') });
+const clean = async (location: string) =>
+  (await fse.exists(fromRoot(location))) && run(`rm -fr ${location}`);
+
+//  Modified from https://stackoverflow.com/questions/45688121/how-to-do-submodule-sparse-checkout-with-git/45689692//45689692
+
+if (argv.clone) {
+  // Clean submodule dirs so we can start fresh
+  await clean(`.git/modules/${submodule}`);
+  await clean(submodule);
+
+  if (process.env.CI) {
+    await run(
+      'git config --global url."https://github.com/".insteadOf "git@github.com:"',
+    );
+  }
+
+  //  Checkout the to-be submodule
+  await run(
+    `git clone --depth=1 --no-checkout -b ${branch} ${repo} ${submodule}`,
+  );
+}
+
+if (argv.absorb) {
+  //  Add as a submodule
+  await run(`git submodule add -b ${branch} --force ${repo} ${submodule}`);
+
+  //  Move the .git dir from $fixture/.git into parent repo's .git
+  await run(`git submodule absorbgitdirs`);
```

Comment:
This potentially quite destructive right? I know that `--absorb` is only set on CI, but I'd feel more confident if this was behind a `process.env.CI` check as well.

Reply:
It's not destructive, but it's behind a flag in case you don't want to do that again if you've done it before.

Reply:
Oh, I read the comment and assumed it was moving things around within `.git`, but it's just a submodule command 🤦